### PR TITLE
ci(trunk): fix docs linting

### PR DESCRIPTION
**Pull request by rriski**
* Remove trailing whitespace and blank line before EOF from docs generation template
* Don't run prettier against the docs
* Remove unused function
* Update go mods after Go 1.22 revert

### CODE REVIEW

File: .trunk/trunk.yaml [hunk 1]
```diff
@@ -22,9 +22,10 @@ lint:
     - prettier@3.2.5
     - yamllint@1.34.0
   ignore:
-    - linters:
-        - yamllint
-        - prettier
+    - linters: [prettier]
+      paths:
+        - docs/docs/**
+    - linters: [yamllint, prettier]
```
Comment (Serpentiel) [inline .trunk/trunk.yaml:28]: why not:

```yaml
- linters:
    - yamllint
    - prettier
```

**Review by Serpentiel — APPROVED** [review]
lgtm, one comment though

**Comment by Serpentiel** [comment]
@rriski feel free to merge, but please reply/action on the comment

File: .trunk/trunk.yaml (hunk 1 above)
Comment (rriski) [reply .trunk/trunk.yaml:28]: I think the [x, y, z] format is more readable in this case. Trunk also suggests that in their docs: https://docs.trunk.io/check/configuration/ignoring-issues#ignoring-multiple-files

**Pull request merged**